src/pip_sync_faster/main.py

```diff
@@ -1,18 +1,75 @@
+import hashlib
+import json
 import sys
 from argparse import ArgumentParser
 from importlib.metadata import version
+from os import environ
+from os.path import abspath
+from pathlib import Path
+from subprocess import CalledProcessError, run
 
 
-def hello_world():
-    return "Hello, world!"
+def get_hash(path):
+    """Return the hash of the given file."""
+    hashobj = hashlib.sha512()
 
+    with open(path, "rb") as file:
+        hashobj.update(file.read())
+
+    return hashobj.hexdigest()
+
+
+def get_hashes(paths):
+    """Return a dict mapping the given files to their hashes."""
+    return {abspath(path): get_hash(abspath(path)) for path in paths}
+
+
+def pip_sync_maybe(src_files, args=None):
+    cached_hashes_path = Path(environ["VIRTUAL_ENV"]) / "pip_sync_faster.json"
+
+    try:
+        with open(cached_hashes_path, "r", encoding="utf-8") as cached_hashes_file:
+            cached_hashes = json.load(cached_hashes_file)
+    except FileNotFoundError:
+        cached_hashes = {}
+
+    hashes = get_hashes(src_files)
+
+    if hashes == cached_hashes:
+        return
+
+    # The hashes did not match the cached ones. This can happen if:
+    #
+    # * One or more of the requirements files has changed: its current hash
+    #   differs from its cached one
+    # * pip-sync-faster was called with a different set of requirements files
+    #   than last time: either one or more of the requirements files are
+    #   missing from the cached hashes, or the cached hashes contain one or
+    #   more requirements files that are missing from the command line args
+
+    run(["pip-sync", *src_files, *(args or [])], check=True)
+
+    # Replace the cached hashes file with one containing the correct hashes for
+    # the requirements files that pip-sync-faster was called with this time.
+    with open(cached_hashes_path, "w", encoding="utf-8") as cached_hashes_file:
+        json.dump(hashes, cached_hashes_file)
+
+
+def entry_point(argv=None):
+    if argv is None:
+        argv = sys.argv[1:]  # pragma: no cover
 
-def entry_point():  # pragma: nocover
     parser = ArgumentParser()
-    parser.add_argument("-v", "--version", action="store_true")
+    parser.add_argument("--version", action="store_true")
+    parser.add_argument("src_files", nargs="*")
 
-    args = parser.parse_args()
+    args = parser.parse_known_args(argv)
 
-    if args.version:
-        print(version("pip-sync-faster"))
+    if args[0].version:
+        print(f"pip-sync-faster, version {version('pip-sync-faster')}")
         sys.exit()
+
+    try:
+        pip_sync_maybe(args[0].src_files, argv[1:])
+    except CalledProcessError as err:
+        sys.exit(err.returncode)
```

Comment:
```suggestion
        sys.exit(err.returncode)
        
if __name__ == '__main__':
    entry_point()
```

This will allow us to run this script directly without fuss.